packages/node/src/__tests__/disable.integration.test.ts

```diff
@@ -1,9 +1,28 @@
-const fetcher = jest.fn()
-jest.mock('../lib/fetch', () => ({ fetch: fetcher }))
-
 import { createTestAnalytics } from './test-helpers/create-test-analytics'
+import { CustomHTTPClient } from '../lib/customhttpclient'
+
+export class CheckFetchClient implements CustomHTTPClient {
```

Comment:
While this works -- Instead of implementing our own custom spy logic to keep track of calls, it would be less boilerplate / more idiomatic to leverage  the existing jest library that has support for mocks/spies and all sorts of goodies. https://jestjs.io/docs/jest-object#jestspyonobject-methodname-accesstype

Ditto elsewhere. You can leverage this pattern to change from success and error between tests... let me know if you want help on that. Also, with that jest automatically clears all mock state between tests, so no need to do any reset.




e.g. 
```ts
import { createTestAnalytics } from './test-helpers/create-test-analytics'
import { createSuccess } from './test-helpers/factories'

describe('disable', () => {
  const mockSend: jest.Mock = jest.fn().mockResolvedValue(createSuccess())

  const checkFetchClient = {
     send: mockSend,
  }

  it('should dispatch callbacks and emit an http request, even if disabled', async () => {
    const analytics = createTestAnalytics({
      disable: true,
    })
    const emitterCb = jest.fn()
    analytics.on('http_request', emitterCb)
    await new Promise((resolve) =>
      analytics.track({ anonymousId: 'foo', event: 'bar' }, resolve)
    )
    expect(emitterCb).toBeCalledTimes(1)
  })

  it('should call fetch if disabled is false', async () => {
    const analytics = createTestAnalytics({
      disable: false,
      customClient: checkFetchClient,
    })
    await new Promise((resolve) =>
      analytics.track({ anonymousId: 'foo', event: 'bar' }, resolve)
    )
    expect(mockSend).toBeCalledTimes(1)
  })
  it('should not call fetch if disabled is true', async () => {
    const analytics = createTestAnalytics({
      disable: true,
      customClient: checkFetchClient,
    })
    await new Promise((resolve) =>
      analytics.track({ anonymousId: 'foo', event: 'bar' }, resolve)
    )
    expect(mockSend).toBeCalledTimes(0)
  })
})

  
```